lib/clAdapters/exchangeService/calendar/index.ts

```diff
@@ -0,0 +1,394 @@
+  'attendeeAddress':                     'Mailbox.EmailAddress',
+  'attendeeName':                        'Mailbox.EmailAddress.Name',
+  'hasAttachments':                      'HasAttachments',
+  'importance':                          'Importance',
+  'iCalUId':                             'ItemId.attributes.Id', //   'iCalUId',
+  'allDay':                              'IsAllDayEvent',
+  'canceled':                            'IsCancelled',
+  // 'isOrganizer':                         'IsOrganizer',
+  'locationName':                        'Location',
+  'organizerName':                       'Organizer.Mailbox.Name',
+  'organizerEmail':                      'Organizer.Mailbox.EmailAddress.Address',
+  'responseRequested':                   'ResponseRequested',
+  'responseStatus':                      'MyResponseType',
+  // 'seriesMasterId':                      'SeriesMasterId',
+  'showAs':                              'ShowAs',
+  'dateTimeStart':                       'Start',
+  'dateTimeEnd':                         'End',
+  'subject':                             'Subject',
+  'type':                                'CalendarItemType',
+  'url':                                 'NetShowUrl',
+  'privacy':                             'Sensitivity'
+};
+
+export interface UserProfile {
+  email: string;
+  emailAfterMapping: string;
+}
+
+export default class ExchangeServiceCalendarAdapter extends ExchangeServiceBaseAdapter {
+  static Configuration = Configuration;
+  static Service = Service;
+
+  // convert the names of the api response data
+  static fieldNameMap = fieldNameMap;
+
+  _config: Configuration;
+  _service: Service;
+  ews: any;
+  soapHeader: any;
+
+  // constructor needs to call super
+  constructor() {
+    super();
+  }
+
+  reset() {
+    delete this._config;
+    delete this._service;
+    return this;
+  }
+
+  async init() {
+    const { credentials }: { credentials: {[k: string]: string} } = this;
+
+    if (!credentials) {
+      throw new Error('credentials required for adapter.');
+    }
+
+    // map json keys to keys used in this library
+    for (const want in credentialMappings) {
+      const alternate = credentialMappings[want];
+      if (!credentials[want]) {
+        credentials[want] = credentials[alternate];
+      }
+    }
+
+    // validate required credential properties
+    Object.keys(credentialMappings)
+      .forEach(prop => {
+        if (!credentials[prop]) {
+          throw new Error(`Property ${prop} required in adapter credentials!`);
+        }
+      });
+
+    this._config  = new ExchangeServiceCalendarAdapter.Configuration(credentials);
+    this._service = new ExchangeServiceCalendarAdapter.Service(this._config);
+
+    await this._service.init();
+
+    const ewsConfig = {
+      username: credentials.username,
+      password: credentials.password,
+      host: credentials.connectUrl
+    };
+
+    this.ews = new EWS(ewsConfig);
```

Comment:
Non-blocking: Can probably combine `this.service` and `this.ews` (`this.service` is supposed to be a wrapper, but if that abstraction doesn't work well, we can prob get rid of it in the base class). Mainly reference for @cynick to account for in his revamp proposal.